src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/BotSegment.cs

```diff
@@ -143,6 +143,40 @@ public void PauseAction()
             }
         }
 
+        public void ProcessValidation()
+        {
+            // Go through each validation and process them. If they have not been started yet, this will also
+            // start them.
+            foreach (var validation in validations)
+            {
+                validation.ProcessValidation(Replay_SegmentNumber);
```

Comment:
Note to myself in my self-review - I need to remove these replay segment numbers, they are not needed by my code I think...

Reply:
leave them in and use them to prefix all log statements with `(#) `... makes it much easier to correlate visually what segment validations are running for

Reply:
Added!